### PR TITLE
Migrate Python Packaging to pyproject.toml

### PR DESCRIPTION
**Pull request by SharonHart**
## Change Description

Modernize packaging by migrating from setup.py to pyproject.toml:

* Dropped all setup.py and setup.cfg
* Added pyproject.toml for all 5 presidio packages
* Used ['flat-layout' automatic package discovery](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#flat-layout), dropping all `tool.setuptools` and `tool.setuptools.packages.find` options 
* Read versions from VERSION file, add one where missing
* Used bdist wheel pakaging, updated release pipeline accordingly. 

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by omri374** [comment]
Looks great. Can you please add the changes to deployment scripts into this PR? or should it come in a subsequent PR?

**Comment by navalev** [comment]
LGTM. Any documentation that needs to change as well?

**Review by navalev — DISMISSED** [review]
tested locally

File: presidio-analyzer/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+]
+test = ["pytest", "ruff"]
+
+[tool.setuptools.package-data]
+presidio_analyzer = [
+    "conf/*",
+]
+
+[tool.setuptools.dynamic]
+version = {file = ["VERSION"]}
```
Comment (RoeyBC) [inline presidio-analyzer/pyproject.toml:52]: @SharonHart is there supposed a line that runs the app? something like 
[tool.flask]
app = "app:app"
?

Comment (SharonHart) [reply presidio-analyzer/pyproject.toml:52]: We run through docker, the old setup.py didnt have that line.

File: presidio-analyzer/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+requires-python = ">= 3.5"
```
Comment (omri374) [inline presidio-analyzer/pyproject.toml:18]: Why not 3.8?

Comment (SharonHart) [reply presidio-analyzer/pyproject.toml:18]: That's what we had in setup.py's, I don't know if this is actually prevents users to install with lower versions (that might work, though we are not supporting them officially)

File: presidio-cli/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+[project.optional-dependencies]
+testing = [
+    "pytest",
+    "ruff",
+]
+
+[tool.setuptools.package-data]
+presidio_cli = ["conf/*.yaml"]
+
+[tool.build_sphinx]
```
Comment (omri374) [inline presidio-cli/pyproject.toml:45]: is sphinx needed here? I think it's a leftover from the project's migration into our repo

**Review by omri374 — APPROVED** [review]

File: presidio-cli/pyproject.toml (hunk 3 above)
Comment (SharonHart) [reply presidio-cli/pyproject.toml:45]: It was introduced along with the cli, after v2. I didn't know if I should drip it, but I think that we can right?

Comment (omri374) [reply presidio-cli/pyproject.toml:45]: Ww are not using sphinx but if it might be an issue let's keep it

Comment (SharonHart) [reply presidio-cli/pyproject.toml]: dropped

**Pull request merged**